pkg/snapshot/store.go

```diff
@@ -0,0 +1,1327 @@
+		}
+		store.FileSystem = &types.StoreFileSystem{
+			Region:               kotsadmVeleroBackendStorageLocation.Spec.Config["region"],
+			Endpoint:             endpoint,
+			ObjectStoreClusterIP: u.Hostname(),
+		}
+		return nil
+	}
+
+	store.Other = &types.StoreOther{
+		Region:   kotsadmVeleroBackendStorageLocation.Spec.Config["region"],
+		Endpoint: endpoint,
+	}
+
+	return nil
+}
+
+// FindBackupStoreLocation will find the backup storage location used by velero
+// kotsadmNamespace is only required in minimal rbac installations. if empty, cluster scope privilages will be needed to detect and validate velero
+func FindBackupStoreLocation(ctx context.Context, kotsadmNamespace string) (*velerov1.BackupStorageLocation, error) {
+	cfg, err := config.GetConfig()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get cluster config")
+	}
+
+	clientset, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create clientset")
+	}
+
+	veleroNamespace, err := DetectVeleroNamespace(ctx, clientset, kotsadmNamespace)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to detect velero namespace")
+	}
+
+	veleroClient, err := veleroclientv1.NewForConfig(cfg)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create velero clientset")
+	}
+
+	backupStorageLocations, err := veleroClient.BackupStorageLocations(veleroNamespace).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to list backupstoragelocations")
+	}
+
+	for _, backupStorageLocation := range backupStorageLocations.Items {
+		if backupStorageLocation.Name == "default" {
+			return &backupStorageLocation, nil
+		}
+	}
+
+	return nil, errors.New("global config not found")
+}
+
+func BuildAWSCredentials(accessKeyID, secretAccessKey string) ([]byte, error) {
+	awsCfg := ini.Empty()
+	section, err := awsCfg.NewSection("default")
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create default section in aws creds")
+	}
+	_, err = section.NewKey("aws_access_key_id", accessKeyID)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create access key")
+	}
+
+	_, err = section.NewKey("aws_secret_access_key", secretAccessKey)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create secret access key")
+	}
+
+	var awsCredentials bytes.Buffer
+	writer := bufio.NewWriter(&awsCredentials)
+	_, err = awsCfg.WriteTo(writer)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to write ini")
+	}
+	if err := writer.Flush(); err != nil {
+		return nil, errors.Wrap(err, "failed to flush buffer")
+	}
+
+	return awsCredentials.Bytes(), nil
+}
+
+func BuildStoreFileSystem(ctx context.Context, clientset kubernetes.Interface, kotsadmNamespace string) (*types.StoreFileSystem, error) {
+	secret, err := clientset.CoreV1().Secrets(kotsadmNamespace).Get(ctx, FileSystemMinioSecretName, metav1.GetOptions{})
+	if err != nil {
```

Comment:
Not for this PR, but I see potential for improvement here.  If `IsNotFound` error returned, we could direct user to installation instructions.  Or at least produce an actionable message.